site/src/components/roadmap/Roadmap.tsx

```diff
@@ -0,0 +1,250 @@
+
+    if (isNaN(startDateA.getTime())) {
+      return 1; // Item A doesn't have a valid date, so it should be placed below item B
+    } else if (isNaN(startDateB.getTime())) {
+      return -1; // Item B doesn't have a valid date, so it should be placed below item A
+    } else {
+      return startDateA.getTime() - startDateB.getTime();
+    }
+  });
+  return sortedData;
+}
+
+
+function formatDate(date: Date): string {
+  if (isNaN(date.getTime())) {
+    return "To be seen";
+  } else {
+    const options: Intl.DateTimeFormatOptions = {
+      day: 'numeric',
+      month: 'long',
+      year: 'numeric',
+    };
+
+
+    const formatter = new Intl.DateTimeFormat(undefined, options);
+    const formattedDateParts = formatter.formatToParts(date);
+
+    const monthNumber = date.getMonth()+1;
+    const quarter = Math.floor((monthNumber-1)/3) + 1;
+    const day = formattedDateParts.find((part) => part.type === 'day')?.value;
+    const month = formattedDateParts.find((part) => part.type === 'month')?.value;
+    const year = formattedDateParts.find((part) => part.type === 'year')?.value;
+    
+    return `${day} ${month} ${year} (Q${quarter})`;
+  }
+}
+
+
+
+function RoadmapCard (props: CardProps) {
+  return <Card className={styles.spacing} style={{ width: "250px", marginBottom: "15px", minHeight: "unset"}}
+  {...props} />
+}
+
+export const Roadmap = ({ title, children }: CalloutProps) => {
+
+  const [jsonData, setJsonData] = useState("");
+  const [roadmapData, setRoadmapData] = useState<any[]>([]);
+  const sortedRoadmapData = sortRoadmapDataByDate(roadmapData);
+  const [searchQuery, setSearchQuery] = useState("");
+
+
+  useEffect(() => {
+   
+    const fetchData = async () => {
+      try {
+        //gets the data
+        const response = await fetch('/api/roadmap');
+        const responseData = await response.json();
+        const jsonString = JSON.stringify(responseData, null, 2);
+        setJsonData(jsonString);
+
+        const items = responseData?.data?.organization?.repository?.projectV2?.items?.nodes;
+        console.log(items); // Check the structure of the extracted items data
+
+        //creates an array of objects with data from github
+        const extractedData = items?.map((item: any) => {
+          const fieldValueNodes = item?.fieldValues?.nodes;
+          const text = getFieldValueByName(fieldValueNodes, "Title");
+          const startDate = getFieldValueByName(fieldValueNodes, "Start Date");
+          const targetDate = getFieldValueByName(fieldValueNodes, "Target Date");
+          const status = getFieldValueByName(fieldValueNodes, "Status");
+          const issueUrl = item?.content?.url;
+
+          return { text, startDate, targetDate, status, issueUrl };
+        });
+
+        
+        //sets the data to retrieved data or null
+        setRoadmapData(extractedData || []);
+        
+      } catch (error) {
+        console.error(error);
+      }
+    };
+
+    fetchData();
```

Comment:
Should we display an error message on the page if no results are found?

Reply:
yup, created a banner that says "No data available", lmk if another component would be better